interview-questions.md

```diff
@@ -9,43 +9,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: It is a package manager for installing libraries, with this we can install gems and also distribute them. I could install a gem that would allow me to use methods that someone else created. There are lots of gems with different uses. I have currently worked with RSPEC and Rails as a gem. 
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on Rails is a framework for creating a web application. I know it uses a Model View Controller structure. The Model is the database, View is what I can see as a user, and Controller is how interactions/responses to users are handled.
 
-Researched answer:
+Researched answer: It is an open-source web app development framework, a gem created in Ruby. Rails makes creating applications easier as it gives a you pre-built structure to work with. Rails uses the Model View Controller architecture. The Model is the data structure, View is the visual side which is what a user interacts with and gives input to. The Controller is how the data is sent to the View and also how the data is handled using CRUD. It also uses Active Record which is how database queries are made in RoR.
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is how information is taken and put into tables. This information is put into columns and rows, the data is stored under similar keys. If I had a table about cars, I would have make, model, year...etc. This information that would follow under would be different but relates to each other. I am sure there are different databases lots which I don't know the names of. I know there are Cloud databases. I haven't worked with databases a whole lot yet. I'm sure there are some I'm familiar with but just don't know the naming for them.
 
-Researched answer:
+Researched answer: It how information is collected and organized in a structure where it stored in a table. It then placed into columns and rows. When information is placed this we can see how it relates to each other. Relational tables are split by attributes which are columns and rows which have the value. These tables contain primary keys which are unique identifier of rows.  This model was created in the 1970s by E.F Codd. 
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are a unique identifier. They are apart of rows, it makes it easier for differentiating between information that is being stored. I know in working with Rails each instance has its own ID when created. It is important because different rows could be easily confused with one another, also we are able to call on that data by its primary key.
 
-Researched answer:
+Researched answer: A primary key must be unique and each table can only have a single primary key. They also cannot contain a Null value.
```

Comment:
Each table row* can have a single primary key! You could get around this issue in an explanation by providing an example